pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -26,13 +30,31 @@ func (ca *cloudCIDRAllocator) performMultiNetworkCIDRAllocation(node *v1.Node, i
 	if err != nil {
 		return nil, fmt.Errorf("node=%s error fetching networks: %v", node.Name, err)
 	}
+
+	// get networks from Node's network-status annotation
+	upStatusNetworks, err := getUpNetworks(node)
+	if err != nil {
+		return nil, err
+	}
+
+	// readyAndUpNetworks is map of networks that are Ready and status==UP
+	readyAndUpNetworks := make(map[string]struct{})
```

Comment:
Thinking about this a bit more, under what condition would a network be in the 'upStatusnetworks' list but its network.Status.Conditions is not ready? In other words, if the network was annotated on the node, the network object must have been in a ready state.

Reply:
I think youre right. If status is up, the network has to be ready. I don't think there is an issue having both checks since it doesnt cost us anything extra to compute.

Do you think we should remove the Network Ready condition from this and only have it be status==up?

Reply:
Yes. I think we should remove it and make the code simpler. I am less worried about the CPU utilization and more worried about code readability. This is much simpler to reason about if we only worry about the network status annotation.